enum.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i])
+        i += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    new_array = []
+    while i < size
+      new_array << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif !arg.nil?
+      my_each { |x| return false unless x == arg }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return true if x == arg }
+    end
+    true
+  end
+
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |x| return true unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return true unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return true if (x =~ arg).nil? }
+    elsif arg.nil?
+      my_each { |x| return true unless x }
+    else
+      my_each { |x| return false if x != arg }
+    end
+    false
+  end
+
+  def my_count(arg = nil)
+    i = 0
+    if block_given?
+      my_each { |x| i += 1 if yield(x) }
+    elsif arg.nil?
+      my_each { |_x| i += 1 }
+    elsif !arg.nil?
+      my_each { |x| i += 1 if arg == x }
+    end
+    i
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    array = []
+    i = 0
+    while i < size
+      array << yield(self[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_inject(accumulator = nil, symbol = nil)
```

Comment:
I feel like maybe you've overcomplicated your my_inject implementation.

Your `my_inject` does not quite work as should with block and symbol.

How `my_inject` works: there are two possible arguments that can be passed. Either the initial accumulator or a symbol or both which you understand.
Here are the following conditions to consider:
- First argument is nil or symbol then the initial accumulator is set to the first element in the array.
- if a block is given and arg1 has value (not symbol) then you yield the block for each element in an array
- if block is given and arg1 is nil then yield block `|num| accumulator = yield(accumulator, num)` from the second element in array `[1..-1].my_each `
- If first arg1 is symbol then you do `|num| accumulator = accumulator.send(arg1, num)` for each element in array from the second element in the array (`[1..-1]`)
- if arg2 is a symbol then you do `|num| accumulator = accumulator.send(arg1, num)` for each element in array (from first)

You return the accumulator at the end of the method. The accumulator is the only thing you'd need to return for this method.

Follow this logic and you shall be fine and pass all tests. 👍